lib/plaid/user.rb

```diff
@@ -112,12 +166,29 @@ def transactions(pending: false, account_id: nil, start_date: nil, end_date: nil
     #
     # Returns self.
     def update(username, password, pin = nil)
```

Comment:
The approach I've taken resembles ActiveRecord, so it should be clear for most Ruby developers.

Just as ActiveRecord, there are some class methods which get you initial objects (`Model.find`, `Model.create`, ...), while modifying operations appear as instance methods in these objects (`save`, `update`, `destroy`, etc.). Data are encapsulated inside the objects and the logic might be tricky, but it's OOP 🔨  Nobody cares about the implementation if the interface is clear and working. Ruby developers also like a bit of "magic". That's exactly what I'm going to give them :)

On the other hand, I agree that the OOP approach introduces extra complexity per se. I like the functional approach with pure functions myself, but Ruby is an odd language for that.